test/extended/router/gatewayapicontroller.go

```diff
@@ -673,3 +760,182 @@ func isOKD(oc *exutil.CLI) (bool, error) {
+	var lbAddress string
+	err := wait.Poll(3*time.Second, 300*time.Second, func() (bool, error) {
+		lbService, err := oc.AdminKubeClient().CoreV1().Services(ingressNamespace).Get(context.Background(), gatewayLbService, metav1.GetOptions{})
+		if err != nil {
+			e2e.Logf("Failed to get the gateway lb service %q: %v, retrying...", gatewayLbService, err)
+			return false, nil
+		}
+
+		createdTime2 := lbService.ObjectMeta.CreationTimestamp
+		if createdTime2 == createdTime1 {
+			e2e.Logf("Trying deleting the original gateway lb service %q, retrying...", gatewayLbService)
+			return false, nil
+		}
+
+		if len(lbService.Status.LoadBalancer.Ingress) == 0 {
+			e2e.Logf("New gateway lb service %q is created, but without lb hostname or ip, retrying...", gatewayLbService)
+			return false, nil
+		}
+
+		if lbService.Status.LoadBalancer.Ingress[0].Hostname != "" {
+			lbAddress = lbService.Status.LoadBalancer.Ingress[0].Hostname
+		} else {
+			lbAddress = lbService.Status.LoadBalancer.Ingress[0].IP
+		}
+		if lbAddress == "" {
+			e2e.Logf("No load balancer address for service %q, retrying", gatewayLbService)
+			return false, nil
+		}
+
+		e2e.Logf("Got load balancer address for service %q: %v", gatewayLbService, lbAddress)
+		return true, nil
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+// used to get the name of a gateway dnsrecord and the index in the dnsrecordList
+func getGWDnsrecordsName(dnsrecordList *operatoringressv1.DNSRecordList, gwName string) (string, int, error) {
+	for index, dnsrecord := range dnsrecordList.Items {
+		if strings.Contains(dnsrecord.ObjectMeta.Name, gwName+"-") {
+			return dnsrecord.ObjectMeta.Name, index, nil
+		}
+	}
+
+	return "", -1, errors.New("Could not get the name of the gw dnsrecord")
+}
+
+// used to wait for the gateway dnsrecord is automatically recreated successfully
+func pollWaitGWDnsrecordsRecreated(oc *exutil.CLI, gwName, ingressNamespace, targetsList1 string) {
+	err := wait.Poll(3*time.Second, 300*time.Second, func() (bool, error) {
+		dnsrecordList, err := oc.AdminIngressClient().IngressV1().DNSRecords(ingressNamespace).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			e2e.Logf("Failed to List DNSRecords in namespace %q: %v, retrying...", ingressNamespace, err)
+			return false, nil
+		}
+
+		_, index, err := getGWDnsrecordsName(dnsrecordList, gwName)
+		if err != nil {
+			e2e.Logf("Failed to get the DNSRecord name for gateway %q in namespace %q: %v, retrying...", gwName, ingressNamespace, err)
+			return false, nil
+		}
+
+		targetsList2 := getSortedString(dnsrecordList.Items[index].Spec.Targets)
+		if targetsList2 != targetsList1 {
+			e2e.Logf("The gateway dnsrecords has not a targetsIP or a different one %s with %s, retrying...", targetsList2, targetsList1)
+			return false, nil
+		}
+
+		e2e.Logf("gwapi dnsrecords targetsList2 is %v", targetsList2)
```

Comment:
I don't think that we need to log current targets since they are the same as expected.